Integrate Edit and Run Time Plugins

ItsMeAshiq opened the pull request:
Integrates the plugins into a single Package.
Moves the common code outside the individual plugins.
Updates the Build configuration to distribute them together as single package.

ItsMeAshiq committed:
Integrate Edit Time & Run Time Plugin Codes

Rename plugin to OpenTypeDef-EditTime
Move common code out

ItsMeAshiq committed:
Merge branch 'LabVIEW-Nation:main' into Integrate-Edit-and-Run-Time-Plugins

ItsMeAshiq committed:
Merge branch 'main' into Integrate-Edit-and-Run-Time-Plugins

# Conflicts:
#	src/code/OpenTypeDef-RunTime.llb/Execute OpenTypeDef-RunTime.vi
#	src/code/OpenTypeDef-RunTime.llb/Find If Constant Clicked Is Typedef.vi
#	src/code/OpenTypeDef-RunTime.llb/Find If Terminal Clicked Is Typedef.vi
#	src/code/OpenTypeDef-RunTime.llb/OpenTypeDef-RunTime.vi

ItsMeAshiq committed:
Fix getting Typedef name with Namespace

Replace Control ref with ControlTerminal ref

ItsMeAshiq committed:
Integrate plugin codes together

ItsMeAshiq committed:
Update Tests

Some tests are failing

ItsMeAshiq committed:
Fix TyeDef Info for Class Controls

Fix class terminal qualified name construction.
Fix Execute VI path.

ItsMeAshiq committed:
Update package settings and enhance descriptions for clarity

ItsMeAshiq committed:
Fix typo in description

ItsMeAshiq committed:
Update README to clarify supported elements and usage

Expanded the README to specify that the tool adds a right-click menu for wires, constants, and terminals carrying TypeDef or Class Object data. Added details about availability in edit and run modes and its usefulness during debugging.

ItsMeAshiq committed:
Remove obsolete manual test file 'Test Main.vi'

ItsMeAshiq committed:
Add usage image to README

Added a new 'Usage.png' image to illustrate features in the README. Updated the README to include a Features section with the image for better documentation.

shivamanikandan commented:
<img width="1177" height="614" alt="image" src="https://github.com/user-attachments/assets/5729283d-4eb8-4dae-8d2b-96d1f15e3c60" />
<img width="1177" height="614" alt="image" src="https://github.com/user-attachments/assets/a13a0840-8417-46ed-b2ed-d1f7e5064e8f" />

shivamanikandan commented:
In the Author names, please add my last name too as mentioned,
<img width="826" height="632" alt="image" src="https://github.com/user-attachments/assets/88eeefda-55d2-4dc1-8572-25d5c50de427" />

shivamanikandan commented:
I didn't find the recent build (.vip) from the repo. So, i have built on my own with your latest code changes and tested the same. All works fine. Great work!

ItsMeAshiq committed:
Update Author Name

Add full name in Author names
Remove redundant code
Rebuild the package with latest changes

ItsMeAshiq commented:
Addressed the comments now.
The tool version is kept at v1.0.0.1 as it was not released.

shivamanikandan commented:
All are good now. Code looks good and build works as expected. Even uninstallation, deletes all our files cleanly from the installed folder.
For documentation, i tested everything with LabVIEW 2022 Q3 (64-Bit).
